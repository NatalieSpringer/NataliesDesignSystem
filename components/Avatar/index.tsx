import React, { useState } from "react";
import { View } from "react-native";
import { AvatarProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Image } from "expo-image";
import { Icon } from "../Icon";

export const Avatar = ({ size = "m", source }: AvatarProps) => {
	const { styles, theme } = useStyles(stylesheet, {
		size
	});

	const [loaded, setLoaded] = useState(false);

	const iconSize = styles.container.size;

	return (
		<View accessible accessibilityRole="image">
			<Image
				contentFit="cover"
				onLoad={() => setLoaded(true)}
				style={[styles.container, { position: "absolute" }]}
				source={source}
				testID="image"
				transition={800}
			/>
			{!loaded && (
				<View style={[styles.container]}>
					<Icon
						color={theme.avatar.icon}
						name="person"
						size={iconSize}
					/>
				</View>
			)}
		</View>
	);
};
