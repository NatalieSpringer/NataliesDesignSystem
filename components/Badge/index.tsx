import React from "react";
import { View } from "react-native";
import { BadgeProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Text } from "../Text";
import { Icon } from "../Icon";

export const Badge = ({ name, text, variant }: BadgeProps) => {
	const { styles, theme } = useStyles(stylesheet, {
		variant
	});

	const foregroundColor = styles.container.textColor;

	return (
		<View style={styles.container}>
			{!!text && (
				<Text color={foregroundColor} size="s" weight="bold">{text}</Text>
			)}
			{!!name && (
				<Icon color={foregroundColor} name={name} size={15} weight="bold" />
			)}
		</View>
	);
};
