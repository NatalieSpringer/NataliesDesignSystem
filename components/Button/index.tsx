import React from "react";
import { TouchableOpacity } from "react-native";
import { ButtonProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Text } from "../Text";

export const Button = ({
	disabled = false,
	onPress,
	text,
	variant
}: ButtonProps) => {
	const { styles } = useStyles(stylesheet, {
		variant
	});

	const containerStyle = styles.container(disabled);
	const { textColor } = containerStyle;

	return (
		<TouchableOpacity
			accessible
			accessibilityLabel={text}
			accessibilityRole="button"
			disabled={disabled}
			onPress={onPress}
			style={containerStyle}
		>
			<Text color={textColor} size="s" weight="bold">
				{text}
			</Text>
		</TouchableOpacity>
	);
};
