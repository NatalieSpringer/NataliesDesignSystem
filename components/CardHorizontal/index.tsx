import React from "react";
import { TouchableOpacity, View } from "react-native";
import { CardHorizontalProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Text } from "../Text";
import { Icon } from "../Icon";

export const CardHorizontal = ({
	description,
	onPress,
	title
}: CardHorizontalProps) => {
	const { styles, theme } = useStyles(stylesheet);

	return (
		<TouchableOpacity onPress={onPress} style={styles.container}>
			<View style={styles.textContainer}>
				<Text size="s" weight="bold">
					{title}
				</Text>
				<Text size="xs">{description}</Text>
			</View>
			<Icon color={theme.card.icon} name="chevronRight" size={16} />
		</TouchableOpacity>
	);
};
