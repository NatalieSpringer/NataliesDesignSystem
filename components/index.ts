export * from "./Avatar";
export * from "./Banner";
export * from "./Button";
export * from "./Icon";
export * from "./Image";
export * from "./Text";
