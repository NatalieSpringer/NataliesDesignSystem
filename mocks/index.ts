import { UnistylesRuntime } from "react-native-unistyles";

export const mockAppTheme = (theme: "light" | "dark") => {
	// @ts-ignore
	jest.spyOn(UnistylesRuntime, "themeName", "get").mockReturnValueOnce(theme);
};
