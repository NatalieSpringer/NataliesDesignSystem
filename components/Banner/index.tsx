import React from "react";
import { View } from "react-native";
import { BannerProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Text } from "../Text";
import { Button } from "../Button";

export const Banner = ({ description, onPress, title }: BannerProps) => {
	const { styles } = useStyles(stylesheet);

	return (
		<View style={styles.container}>
			<Text color="black" size="s" weight="bold">
				{title}
			</Text>
			<Text color="black" size="xs">
				{description}
			</Text>
			<View style={styles.button}>
				<Button onPress={onPress} text="Button" variant="primary" />
			</View>
		</View>
	);
};
