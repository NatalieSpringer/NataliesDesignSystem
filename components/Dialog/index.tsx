import React from "react";
import { View } from "react-native";
import { DialogProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Button } from "../Button";

export const Dialog = ({ buttons, variant }: DialogProps) => {
	const { styles } = useStyles(stylesheet, {
		variant
	});

	return (
		<View style={styles.container}>
			{variant === "primary" ? (
				<View style={styles.buttonContainer}>
					<View style={styles.primaryButton}>
						<Button {...buttons[0]} variant="secondary" />
					</View>
					<View style={styles.primaryButton}>
						<Button {...buttons[1]} variant="primary" />
					</View>
				</View>
			) : (
				<View style={styles.secondaryButton}>
					<Button {...buttons[0]} variant="secondary" />
					<Button {...buttons[1]} variant="secondary" />
					<Button {...buttons[2]} variant="primary" />
				</View>
			)}
		</View>
	);
};
