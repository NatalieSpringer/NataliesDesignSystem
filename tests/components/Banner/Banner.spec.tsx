import { Banner } from "../../../components";
import { render } from "@testing-library/react-native";
import { mockAppTheme } from "../../../mocks";

describe("Banner", () => {
	["dark", "light"].forEach((theme) => {
		beforeEach(() => {
			mockAppTheme(theme as "light" | "dark");
		});
		it(`should render correctly in ${theme}`, () => {
			const onPress = jest.fn();
			const tree = render(
				<Banner
					description="Description"
					onPress={onPress}
					title="Title"
				/>
			).toJSON();

			expect(tree).toMatchSnapshot();
		});
	});
});
