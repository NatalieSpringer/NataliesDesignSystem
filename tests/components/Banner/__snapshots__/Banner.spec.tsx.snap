// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should render correctly in dark 1`] = `
<View
  style={
    {
      "backgroundColor": "#2F3036",
      "borderRadius": 16,
      "padding": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "black",
          "fontFamily": "SFProText-Regular",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        false,
        {
          "textAlign": "left",
        },
        {
          "lineHeight": 24,
        },
      ]
    }
  >
    Title
  </Text>
  <Text
    style={
      [
        {
          "color": "black",
          "fontFamily": "SFProText-Regular",
          "fontSize": 14,
        },
        false,
        {
          "textAlign": "left",
        },
        {
          "lineHeight": 21,
        },
      ]
    }
  >
    Description
  </Text>
  <View
    style={
      {
        "marginTop": 20,
        "width": "50%",
      }
    }
  >
    <View
      accessibilityLabel="Button"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 122, 255, 1)",
          "borderRadius": 12,
          "justifyContent": "center",
          "minHeight": 40,
          "opacity": 1,
          "textColor": "white",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "black",
              "fontFamily": "SFProText-Regular",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "white",
            },
            {
              "textAlign": "left",
            },
            {
              "lineHeight": 24,
            },
          ]
        }
      >
        Button
      </Text>
    </View>
  </View>
</View>
`;

exports[`Banner should render correctly in light 1`] = `
<View
  style={
    {
      "backgroundColor": "#2F3036",
      "borderRadius": 16,
      "padding": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "black",
          "fontFamily": "SFProText-Regular",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        false,
        {
          "textAlign": "left",
        },
        {
          "lineHeight": 24,
        },
      ]
    }
  >
    Title
  </Text>
  <Text
    style={
      [
        {
          "color": "black",
          "fontFamily": "SFProText-Regular",
          "fontSize": 14,
        },
        false,
        {
          "textAlign": "left",
        },
        {
          "lineHeight": 21,
        },
      ]
    }
  >
    Description
  </Text>
  <View
    style={
      {
        "marginTop": 20,
        "width": "50%",
      }
    }
  >
    <View
      accessibilityLabel="Button"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 122, 255, 1)",
          "borderRadius": 12,
          "justifyContent": "center",
          "minHeight": 40,
          "opacity": 1,
          "textColor": "white",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "black",
              "fontFamily": "SFProText-Regular",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "white",
            },
            {
              "textAlign": "left",
            },
            {
              "lineHeight": 24,
            },
          ]
        }
      >
        Button
      </Text>
    </View>
  </View>
</View>
`;
